### DISCUSSION TITLE
Corrupt HDD - What to do next

### DISCUSSION THREAD

**Discussion by bitsam** [raspiblitz/raspiblitz#4753] [Q&A]
Hi Community. I'm looking for some help on how to recover my lightning node. I noticed a few weeks ago that all channels went offline, so I went into the raspiblitz and started a node restart. It never came back online after the reboot. I was traveling and didn't have time to troubleshoot this, however when I finally plugged in a monitor and watched the post/boot. It was doing the filesystem check and failing. I've plugged the SSD into a Linux server and can see all the folders, however when I tried copying the data, I got the error that the system/user on the other linux system does not have permissions to access the data. I am a linux newb, so not that confident in what to do next. In a past life i have used chmod to change permissions. 

If I wind back a little, in reading another post, the steps to recover are use my seed words and a channel backup. I don't have a recent channel backup, as I closed and re-opened a few channels since my last backup. If I am able to pull files off the disk/SSD, what files am I looking for to use / copy back after I have re-built the new node and filesystem on the external disk/SSD?

I am using a Rpi 4 with v1.11.2 and a 1TB SSD.

**Comment by bitsam** [comment]
I was able to view and copy all the files from /lnd/data/chain/bitcoin/mainnet/ . Can I create my own migration file or lndrescue file to use when rebuilding the node?

If not, what files can I copy from my old disk, and replace on the new node to assist with the restore?

**Comment by rootzoll** [comment]
> Can I create my own migration file or lndrescue file to use when rebuilding the node?

In general I would concentrate on the LND wallet - which is not the big migration file (containing all the app-data) but just the LND oder CLN rescue file - which is basically a tar zip of the LND data directory.

Here you can find the script creating such a LND rescue file:
https://github.com/raspiblitz/raspiblitz/blob/667e0b9364a406475b9771bfe5c491a25dfd305e/home.admin/config.scripts/lnd.backup.sh#L134-L135

If you want to manually create a lnd rescue file from a recovered lnd directory you can try the following:
```
# Add LND version info to the lnd directory
echo "v0.17.5-beta" > version.info

# Navigate to the parent directory of lnd
cd ..

# Create the compressed backup
tar -zcvf lnd-rescue.tar.gz lnd

# Calculate MD5 checksum
md5checksum=$(md5sum lnd-rescue.tar.gz | awk '{print $1}')

# Rename the backup file to include checksum
mv lnd-rescue.tar.gz lnd-rescue-${md5checksum}.tar.gz
```

Now you can try to upload that rescue file during SSH setup of a fresh RaspiBlitz (choose LND and later I have LND rescue file).

**Comment by bitsam** [comment]
Thank you @rootzoll , I have a full blockchain download and have created the lnd-rescue file.  I can't recall where I was meant to upload the lnd-rescue file to. Can you confirm where I need to upload the lnd-rescue file?  I started a new setup on my node, so skipped the recovery setup (over a week ago). Should I put in a newly formatted SD with the latest reaspiblitz image to get back to the recover/rescue menu?

**Comment by rootzoll** [comment]
Put a fresh sd card in and SSH in. Choose "Other Options" and "Fresh Setup". If you wanna keep the Blockchain you can do so during the process - all other data gets deleted on fresh setup. Then choose LND and that you already have a wallet, then choose to upload LND rescue file and let the setup run thru.

**Comment by bitsam** [comment]
I am still having trouble getting my node back online. I downloaded a full copy of the blockchain on another computer and when i copied it across, the raspiblitz errors out of the setup process with the error to reindex the blockchain. This looks to be a 2 week process just to reindex. Is this normal, that after copying the blocks, index, and chainstate from an up-to-date Bitcoin Core node the raspiblitz still has to reindex the chainstate, or am I doing something wrong?

EDIT. My source system had a nested folder and it was copying the blockchain to an incorrect folder layout on the raspiblitz. I now have a full copy of the blockchain, but now my lndrescue file fails to provision at the copying LND macroons step.
